### ISSUE TITLE
Adding Gemma 3n support?

### ISSUE THREAD

**Issue by dwgb93** [axolotl-ai-cloud/axolotl#2850]
### ⚠️ Please check that this feature request hasn't been suggested before.

- [x] I searched previous [Ideas in Discussions](https://github.com/axolotl-ai-cloud/axolotl/discussions/categories/ideas) didn't find any similar feature requests.
- [x] I searched previous [Issues](https://github.com/axolotl-ai-cloud/axolotl/labels/enhancement) didn't find any similar feature requests.

### 🔖 Feature description

I saw you guys were mentioned in the [Gemma 3n Developer's Guide](https://developers.googleblog.com/en/introducing-gemma-3n-developer-guide/) and wanted to check if you were tracking to add support for the [family of MatFormer models](https://huggingface.co/collections/google/gemma-3n-685065323f5984ef315c93f4) recently released by Google.

### ✔️ Solution

I tried using the [gemma3 examples](https://github.com/axolotl-ai-cloud/axolotl/tree/main/examples/gemma3) as a starting point, and after upgrading transformers to [v4.53.0](https://github.com/huggingface/transformers/releases/tag/v4.53.0) and installing timm, I still got opaque errors like 

> ) is not supported. Currently, only the following modules are supported: `torch.nn.Linear`, `torch.nn.Embedding`, `torch.nn.Conv1d`, `torch.nn.Conv2d`, `torch.nn.Conv3d`, `transformers.pytorch_utils.Conv1D`, `torch.nn.MultiheadAttention.`.

I'd love it if, like most other models I've finetuned with axolotl, it just worked.

### ❓ Alternatives

Unsloth says they have [Gemma 3n support](https://docs.unsloth.ai/basics/gemma-3n-how-to-run-and-fine-tune), but their page just says "Gemma 3n Notebooks coming soon! Stay tuned."

### Acknowledgements

- [x] My issue title is concise, descriptive, and in title casing.
- [x] I have searched the existing issues to make sure this feature has not been requested yet.
- [x] I have provided enough information for the maintainers to understand and evaluate this request.

**Comment by NanoCode012** [comment]
Thanks. I got a simple config running. However, the loss is very unusually high. Will need to look into this further.

Your issue can be fixed with:
```diff
-lora_target_linear: true
```

and manually target the layers you want
```
lora_target_modules:
  - q_proj
  - k_proj
```

**Comment by dwgb93** [comment]
@NanoCode012 thanks for your help!
I'm still getting stuck though.

I was getting an error about gemma3n not being supported. I upgraded cut-cross-entropy, per your pull request, and tried again using the example config file you made. This gave met the longest error message I've ever seen.

It printed out the entire gemma3n architecture, before ending:
> ) got multiple values for keyword argument 'return_dict'

Any idea what that means?

**Comment by NanoCode012** [comment]
Hey @dwgb93 , I would say to wait a bit for us to experiment further with this. There's quite a few things we need to fix first.

> ) got multiple values for keyword argument 'return_dict'

Do you have more logs for this?

If you want to experiment, the linked PR has a sample config.

**Comment by NanoCode012** [comment]
I'm not sure how but I triggered this error too with minimal changes.The fix can be found here for now till I figure out if anything else is wrong https://github.com/axolotl-ai-cloud/ml-cross-entropy/pull/5

**Comment by dwgb93** [comment]
Nice! I'll mess around with what you've provided, but I'll let you guys cook in the meantime.

Thanks for your help!
